models/vehicle.rb

```diff
@@ -118,8 +118,8 @@ def self.create(hash)
     end
 
     def need_matrix_time?
-      cost_time_multiplier != 0 || cost_late_multiplier && cost_late_multiplier != 0 || cost_setup_time_multiplier != 0 ||
-        !rests.empty? || maximum_ride_time || duration || overall_duration
+      cost_time_multiplier.positive? || timewindow.end || cost_late_multiplier&.positive? ||
+        cost_setup_time_multiplier.positive? || !rests.empty? || maximum_ride_time || duration || overall_duration
     end
```

Comment:
Don't we always need time_matrix ?

Reply:
VROOM only tolerate a single dimension.

If we don't need one dimension with HERE matrix, we can save a large amount of time